pkg/kubecost/allocation.go

```diff
@@ -2260,8 +2260,39 @@ func (asr *AllocationSetRange) Accumulate() (*AllocationSet, error) {
 	return allocSet, nil
 }
 
-// TODO accumulate into lower-resolution chunks of the given resolution
-// func (asr *AllocationSetRange) AccumulateBy(resolution time.Duration) *AllocationSetRange
+// AccumulateBy sums AllocationSets based on the resolution given. The resolution given is subject to the scale used for the AllocationSets.
+// If the requested resolution is smaller than the window of an AllocationSet then the resolution will default to the duration of a set.
+// Resolutions larger than the duration of the entire AllocationSetRange will default to the duration of the range.
+func (asr *AllocationSetRange) AccumulateBy(resolution time.Duration) (*AllocationSetRange, error) {
+	allocSetRange := &AllocationSetRange{allocations: []*AllocationSet{}}
```

Comment:
Small suggestion: we could use `NewAllocationSetRange` here.

Today, this matches that implementation, but imagine that the type changes and something important is added to the constructor. (Like, for instance, something could be added to that type that would nil panic if not properly instantiated.) We'd want that extra logic to come along.

Reply:
Makes sense. The initial concern was that initiating a new AllocationSet would result in an incorrect accumulation based on the window start/end parameters we're forced to include in the call.  However I verified that AllocationSet's IsEmpty() will still return empty with non-nil window durations when accumulating so fixing this shouldn't cause issues

Fixed!

Reply:
Initial concern was resolved so I'm in agreement, fixed!